packages/web-runtime/tests/components/UserMenu.spec.js

```diff
@@ -0,0 +1,76 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import Vuex from 'vuex'
+import { Store } from 'vuex-mock-store'
+import UserMenu from 'web-runtime/src/components/UserMenu.vue'
+import stubs from '../../../../tests/unit/stubs'
+
+const localVue = createLocalVue()
+localVue.use(Vuex)
+jest.useFakeTimers()
+
+describe('User Menu component', () => {
+  describe('when userId is set', () => {
+    const wrapper = shallowMount(UserMenu, {
+      localVue,
+      stubs,
+      propsData: {
+        userId: 'einstein',
+        userDisplayName: 'Albert Einstein'
+      }
+    })
+    it('displays the avatar', () => {
+      expect(wrapper.find('.oc-topbar-personal-avatar').exists()).toBe(true)
+      expect(wrapper.find('.oc-topbar-personal-label').text()).toEqual('Albert Einstein')
+    })
+    it('displays the profile link', () => {
+      expect(wrapper.find('#oc-topbar-account-manage').exists()).toBe(true)
+    })
+    it('displays the logout button', () => {
+      expect(wrapper.find('#oc-topbar-account-logout').exists()).toBe(true)
+    })
+  })
+  describe('when userId is empty', () => {
+    const wrapper = shallowMount(UserMenu, {
+      localVue,
+      stubs,
+      propsData: {
+        userId: '',
+        userDisplayName: 'Albert Einstein'
+      }
+    })
+    it('does not display the avatar', () => {
+      expect(wrapper.find('.oc-topbar-personal-avatar').exists()).toBe(false)
+      expect(wrapper.find('.oc-topbar-personal-label').exists()).toBe(false)
+    })
+    it('does not display the profile link', () => {
+      expect(wrapper.find('#oc-topbar-account-manage').exists()).toBe(false)
+    })
+    it('does not display the logout button', () => {
+      expect(wrapper.find('#oc-topbar-account-logout').exists()).toBe(false)
+    })
+  })
+  describe('methods', () => {
+    const store = new Store({})
+    beforeEach(() => {
+      store.reset()
+    })
+    const wrapper = shallowMount(UserMenu, {
+      localVue,
+      stubs,
+      propsData: {
+        userId: 'einstein',
+        userDisplayName: 'Albert Einstein'
+      },
+      mocks: {
+        $store: store
+      }
+    })
+
+    it('does logout', () => {
+      wrapper.find('#oc-topbar-account-logout').vm.$emit('click')
```

Comment:
Please simulate a click with `wrapper.find('#oc-topbar-account-logout').trigger('click')`

Since we're using an `oc-button` component inside the user menu, a `shallowMount` will not be sufficient then. Look at the `SharedWithMe` unit tests for examples how to mount and also include the design system.

Reply:
done, the trick was to use the design system with `localVue.use(DesignSystem)`, use `mount` instead of `shallowMount` and not stub `oc-button` but only unused elements
```
      stubs: {
        'router-link': true,
        translate: true,
        'oc-drop': true,
        'avatar-image': true
      },
```

CC @kiranparajuli589 @saw-jan FYI ^

Reply:
thank you @individual-it. this fix my blocker in notification component